aptos-move/framework/aptos-names/sources/domains.move

```diff
@@ -133,7 +138,7 @@ module aptos_names::domains {
 
     /// A wrapper around `register_name` as an entry function.
     /// Option<String> is not currently serializable, so we have these convenience methods
-    public entry fun register_domain(sign: &signer, domain_name: String, num_years: u8) acquires NameRegistryV1, RegisterNameEventsV1, SetNameAddressEventsV1 {
+    public entry fun register_domain(sign: &signer, domain_name: String, num_years: u8, signature: vector<u8>) acquires NameRegistryV1, RegisterNameEventsV1, SetNameAddressEventsV1 {
```

Comment:
I am very confused: both `register_domain` and `register_domain_with_signature` seem to do the same thing...

Why two different functions?

Reply:
see my comments above! lmk what you think : )